apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -459,6 +462,38 @@ class LivechatClass {
 			extraData,
 		});
 
+		if (isSingleContactEnabled()) {
+			let { contactId } = visitor;
+			if (!contactId) {
+				const visitorContact = await LivechatVisitors.findOne<Pick<ILivechatVisitor, 'contactId'>>(visitor._id, {
+					projection: { contactId: 1 },
+				});
+				contactId = visitorContact?.contactId;
+			}
```

Comment:
I assume this is for covering the cases where visitor is called with less data, but i wonder if it wouldn't be better to ensure visitor has always the complete data and avoid this db call 🤔 wdyt?

Reply:
Pierre suggested the same. I've added the contactId on the projection of every place I could find to ensure that we're passing the parameters. I chose to leave this verification in case we miss some spots.